demos/smart_classroom_demo_gapi/include/custom_kernels.hpp

```diff
@@ -0,0 +1,149 @@
+    static const char* tag() {
+        return "custom.get_action_detection_result_state_params";
+    }
+};
+
+template<> struct cv::detail::CompileArgTag<TrackerParamsPack> {
+    static const char* tag() {
+        return "custom.get_recognition_result_state_params";
+    }
+};
+
+namespace custom {
+    G_API_OP( FaceDetectorPostProc
+            , <cv::GArray<detection::DetectedObject>( cv::GMat
+                                                    , cv::GMat
+                                                    , detection::FaceDetectionKernelInput)>
+            , "custom.fd_postproc") {
+        static cv::GArrayDesc outMeta( const cv::GMatDesc &
+                                     , const cv::GMatDesc &
+                                     , const detection::FaceDetectionKernelInput &) {
+            return cv::empty_array_desc();
+        }
+    };
+
+    G_API_OP( GetRectFromImage
+            , <cv::GArray<cv::Rect>(cv::GMat)>
+            , "custom.get_rect_from_image") {
+        static cv::GArrayDesc outMeta(const cv::GMatDesc &) {
+            return cv::empty_array_desc();
+        }
+    };
+
+    G_API_OP( GetRectsFromDetections
+            , <cv::GArray<cv::Rect>(cv::GArray<detection::DetectedObject>)>
+            , "custom.get_rects_from_detection") {
+        static cv::GArrayDesc outMeta(const cv::GArrayDesc &) {
+            return cv::empty_array_desc();
+        }
+    };
+
+    G_API_OP( PersonDetActionRecPostProc
+            , <cv::GArray<DetectedAction>( cv::GMat, cv::GMat
+                                         , cv::GMat, cv::GMat
+                                         , cv::GMat, cv::GMat
+                                         , cv::GMat, cv::GMat
+                                         , ActionDetectionKernelInput)>
+            , "custom.person_detection_action_recognition_postproc") {
+        static cv::GArrayDesc outMeta( const cv::GMatDesc &, const cv::GMatDesc &
+                                     , const cv::GMatDesc &, const cv::GMatDesc &
+                                     , const cv::GMatDesc &, const cv::GMatDesc &
+                                     , const cv::GMatDesc &, const cv::GMatDesc &
+                                     , const ActionDetectionKernelInput &) {
+            return cv::empty_array_desc();
+        }
+    };
+
+    G_API_OP( AlignFacesForReidentification
+            , <cv::GMat( cv::GMat
+                       , cv::GArray<cv::GMat>
+                       , cv::GArray<cv::Rect>)>
+            , "custom.align_faces_for_reidentification") {
+        static cv::GMatDesc outMeta( const cv::GMatDesc & in
+                                   , const cv::GArrayDesc &
+                                   , const cv::GArrayDesc &) {
+            return in;
+        }
+    };
+
+    template<typename TO, typename S, typename TR> using rout = std::tuple<TO, TO, S, S, TR>;
+    template<typename T> using tuple_five = std::tuple<T, T, T, T, T>;
```

Comment:
Somewhere in the samples I've used `duo<T> = tuple<T,T>`, `trio<T> = tuple<T,T,T`, and `quad<T> = tuple<T,T,T,T>`.

I believe `penta` stands for 5. :)